### PR TITLE
Support multiple output plugins of the same type

### PR DESCRIPTION
**Pull request by KarolisL**
We need to write (replicate) the same metrics to 2 different influxdb servers but that's not possible with current implementation

This PR defines telegraf::output type which can be used to fulfill our needs.

Any comments/suggestions welcome. :-)

### CODE REVIEW

File: manifests/output.pp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#   Hash. Some inputs take multiple sections.
+#
+define telegraf::output (
+  $plugin_type = $name,
+  $options     = undef,
+  $suboptions  = undef,
+) {
+  include telegraf
+
+  notice("Creating: ${name} :: ${plugin_type}")
```
Comment (yankcrime) [inline manifests/output.pp]: I don't think this `notice` is necessary.

Comment (KarolisL) [reply manifests/output.pp]: Thx for noticing. Removed that.

File: manifests/output.pp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# == Define: telegraf::input
```
Comment (yankcrime) [inline manifests/output.pp]: `::telegraf::output`

File: manifests/output.pp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#
+# A Puppet wrapper for discrete Telegraf input files
```
Comment (yankcrime) [inline manifests/output.pp]: 'output'

File: manifests/output.pp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#
+# === Parameters
+#
+# [*options*]
+#   Hash. Plugin options for use the the input template.
+#
+# [*sections*]
+#   Hash. Some inputs take multiple sections.
```
Comment (yankcrime) [inline manifests/output.pp]: 'outputs'

File: manifests/output.pp [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# == Define: telegraf::input
+#
+# A Puppet wrapper for discrete Telegraf input files
+#
+# === Parameters
+#
+# [*options*]
+#   Hash. Plugin options for use the the input template.
```
Comment (yankcrime) [inline manifests/output.pp]: 'output'

File: manifests/output.pp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  if $options {
+    validate_hash($options)
+  }
+
+  if $sections {
+    validate_hash($sections)
+  }
+
+  Class['::telegraf::config']
+  ->
```
Comment (mxjessie) [inline manifests/output.pp]: If you move these chaining arrows (lines 29 & 33) to prepend the same line as their chained directives (e.g. `->\nfile {...` becomes `-> file {...`), I think tests will pass on this PR.

Comment (yankcrime) [reply manifests/output.pp]: 👍

**Comment by KarolisL** [comment]
Tests still fail with cryptic error message. Have I made some other mistake?

File: manifests/output.pp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# [*options*]
+#   Hash. Plugin options for use the the output template.
+#
+# [*sections*]
+#   Hash. Some outputs take multiple sections.
+#
+define telegraf::output (
+  $plugin_type = $name,
+  $options     = undef,
+  $suboptions  = undef,
```
Comment (mxjessie) [inline manifests/output.pp:16]: need to declare ```$sections = undef,``` (e.g.) here

**Comment by mxjessie** [comment]
I've got an open PR to fix the incomprehensible ```rake lint``` errors. If you declare ```$sections``` and pull to the latest master, I think tests will pass.

**Comment by yankcrime** [comment]
@KarolisL If you want to make the changes as @mxjessie suggests then this should be good to go.

**Comment by KarolisL** [comment]
Thanks!

Should I squash the commits?

**Comment by yankcrime** [comment]
@KarolisL Please - I can 'squash and merge' but it would be best if you cleaned up your history 👍

**Comment by KarolisL** [comment]
@yankcrime done

**Comment by yankcrime** [comment]
Thanks @KarolisL !

**Pull request merged**